helpers/manifest-parser.js

```diff
@@ -0,0 +1,325 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const ALLOWED_DISPLAY_VALUES = [
+  'fullscreen',
+  'standalone',
+  'minimal-ui',
+  'browser'
+];
+
+const ALLOWED_ORIENTATION_VALUES = [
+  'any',
+  'natural',
+  'landscape',
+  'portrait',
+  'portrait-primary',
+  'portrait-secondary',
+  'landscape-primary',
+  'landscape-secondary'
+];
+
+function parseString(raw, trim) {
+  let value;
+  let warning;
+
+  if (typeof raw === 'string') {
+    value = trim ? raw.trim() : raw;
+  } else {
+    if (raw !== undefined) {
+      warning = 'ERROR: expected a string.';
+    }
+    value = undefined;
+  }
+
+  return {
+    raw,
+    value,
+    warning
+  };
+}
+
+function parseURL(raw) {
+  // TODO: resolve url using baseURL
+  // var baseURL = args.baseURL;
+  // new URL(parseString(raw).value, baseURL);
+  return parseString(raw, true);
+}
+
+function parseColor(raw) {
+  let color = parseString(raw);
+
+  if (color.value === undefined) {
+    return color;
+  }
+
+  // TODO(bckenny): validate color, but for reals
+  // possibly pull in devtools/front_end/common/Color.js
```

Comment:
as stated here, planning on bringing in e.g. devtools frontend color parsing code rather than dragging in all of jsdom just to do some color parsing code